test/test_common/utility.cc

```diff
@@ -26,18 +26,21 @@
 #include "test/test_common/printers.h"
 
 #include "absl/strings/string_view.h"
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
 #include "gtest/gtest.h"
 
 using testing::GTEST_FLAG(random_seed);
 
 namespace Envoy {
 
-static const int32_t SEED = std::chrono::duration_cast<std::chrono::nanoseconds>(
-                                std::chrono::system_clock::now().time_since_epoch())
-                                .count();
+int32_t getSeed() {
+  static int32_t seed = absl::ToUnixNanos(absl::Now());
```

Comment:
const?

Reply:
It is necessary to add const to value type?

Reply:
More good reading about the static initialization "fiasco" referenced in the AddressSanitizer report is here:

https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use

My personal rule of thumb: avoid class statics and pre-main function-calls wherever possible and programs will be easier to reason about and more portable. Compile-time initialized scalars, arrays, and structs are fine.

Re the "const" question; that's covered in https://google.github.io/styleguide/cppguide.html#Use_of_const